test/testscripts/scale_alpha_discrete.jl

```diff
@@ -0,0 +1,19 @@
+using DataFrames, Gadfly
+
+set_default_plot_size(21cm, 8cm)
+
+D = DataFrame(x=1:6, y=[0.39, 0.26, 0.31, 0.43, 0.1, 0.78], Shape=repeat(["a","b","c"], outer=2))
+coord = Coord.cartesian(xmin=0, xmax=7, ymin=0, ymax=1.0)
+p1 = plot(D, x=:x, y=:y, color=:x,  coord,
+    Scale.color_discrete, Geom.point, Geom.hair,
+    Guide.title("Scale.color_discrete, Theme(alphas=[0.5])"),
+    Theme(alphas=[0.5], discrete_highlight_color=identity,
+        point_size=2mm)
+)
+p2 = plot(D, x=:x, y=:y, alpha=:x, shape=:Shape, coord,
+    Scale.alpha_discrete, Geom.point, Geom.hair,
+    Guide.title("Scale.alpha_discrete, Theme(default_color=\"green\")"),
+    Theme(default_color="green",   discrete_highlight_color=c->"gray",
+        point_size=2mm, alphas=[0.0,0.2,0.4,0.6,0.8,1.0])
+)
+hstack(p1,p2)
```

Comment:
both of these tests are near identical to the docs, the only difference being the hard-coded random numbers.  seems redundant.  ideally i like to see real data, say from RDatasets, used in the docs that is particularly pretty, and very simple made up cases for the tests with focus on execution speed (we time out sometimes) and code coverage.  this is a bit of a nit-pick, but can you at least make the docs and the tests less redundant?

Reply:
I like this example because it illustrates e.g. "color aesthetic, fixed alpha" and "alpha aesthetic, fixed color".  If I used an example from RDatasets to illustrate the `alpha` aesthetic, I would use something like the `Diamonds` dataset (53940 points).  In my next PR I'm going to halve the time it takes to plot that: `Geom.point` currently suffers from using a new context for every point.